project-api/ballerina-projects/src/main/java/io/ballerina/projects/balo/BaloFiles.java

```diff
@@ -0,0 +1,200 @@
+
+        if (!Files.exists(packageJsonPath)) {
+            throw new RuntimeException("package.json does not exists:" + packageJsonPath);
+        }
+        // Load `package.json`
+        PackageJson packageJson;
+        try {
+            packageJson = gson.fromJson(Files.newBufferedReader(packageJsonPath), PackageJson.class);
+        } catch (IOException e) {
+            throw new RuntimeException("package.json does not exists:" + packageJsonPath);
+        }
+        validatePackageJson(packageJson);
+        // Create Package
+        Package tomlPackage = new Package();
+        tomlPackage.setOrg(packageJson.getOrganization());
+        tomlPackage.setName(packageJson.getName());
+        tomlPackage.setVersion(packageJson.getVersion());
+        tomlPackage.setLicense(packageJson.getLicenses());
+        tomlPackage.setAuthors(packageJson.getAuthors());
+        tomlPackage.setRepository(packageJson.getSourceRepository());
+        tomlPackage.setKeywords(packageJson.getKeywords());
+
+        ballerinaToml.setPkg(tomlPackage);
+        return ballerinaToml;
+    }
+
+    private static void validatePackageJson(PackageJson packageJson) {
+        if (packageJson.getOrganization() == null || "".equals(packageJson.getOrganization())) {
+            throw new RuntimeException("'organization' does not exists in 'package.json'");
+        }
+        if (packageJson.getName() == null || "".equals(packageJson.getName())) {
+            throw new RuntimeException("'name' does not exists in 'package.json'");
+        }
+        if (packageJson.getVersion() == null || "".equals(packageJson.getVersion())) {
+            throw new RuntimeException("'version' does not exists in 'package.json'");
+        }
+    }
+
+    private static ModuleData loadModule(Path modulePath) {
+        // check module path exists
+        if (!Files.exists(modulePath)) {
+            throw new RuntimeException("module does not exists:" + modulePath);
+        }
+
+        String moduleName = String.valueOf(modulePath.getFileName());
+        if (moduleName.contains(".")) { // not default module
+            moduleName = moduleName.split("\\.")[1];
+            moduleName = moduleName.replace("/", "");
+        }
+
+        // validate moduleName
+        if (!ProjectUtils.validateModuleName(moduleName)) {
+            throw new RuntimeException("Invalid module name : '" + moduleName + "' :\n" +
+                    "Module name can only contain alphanumerics, underscores and periods " +
+                    "and the maximum length is 256 characters");
+        }
+        List<DocumentData> srcDocs = loadDocuments(modulePath);
+        List<DocumentData> testSrcDocs = Collections.emptyList();
+
+        // TODO Read Module.md file. Do we need to? Balo creator may need to package Module.md
+        return ModuleData.from(modulePath, srcDocs, testSrcDocs);
+    }
+
+    private static List<DocumentData> loadDocuments(Path dirPath) {
```

Comment:
This method is already there in ProjectFiles class. Shall we make use of that one without duplicating it here?

Reply:
Then we have to make that method `public`. Shall we move these common methods to a `util` class?

Reply:
Added